lib/datadog/appsec/contrib/rack/request_middleware.rb

```diff
@@ -75,6 +75,16 @@ def call(env)
 
             _response_return, response_response = Instrumentation.gateway.push('rack.response', gateway_response)
 
+            result = scope.processor_context.extract_schema
+
+            if result
+              scope.processor_context.events << {
+                trace: scope.trace,
+                span: scope.service_entry_span,
+                waf_result: result,
+              }
+            end
```

Comment:
I'm confused by this, the extracted schema is a waf result, and stored in the events list?